Author: Andreas Beckmann <anbe@debian.org>
Bug-Debian: http://bugs.debian.org/717361
Origin: other, https://projects.archlinux.org/svntogit/packages.git/plain/trunk/nvidia-linux-3.10.patch?h=packages/nvidia-304xx&id=787962153c92600e8a6d0756f026e6c9a87e48f8
Description: adjust for kernel 3.10 i2c interface changes
 based on the archlinux patch (author unknown)
 restored backward compatibility by wrapping all changes with
 KERNEL_VERSION(3,10,0) checks

--- a/nv-i2c.c
+++ b/nv-i2c.c
@@ -311,8 +311,10 @@ void* NV_API_CALL nv_i2c_add_adapter(nv_state_t *nv, NvU32 port)
 BOOL NV_API_CALL nv_i2c_del_adapter(nv_state_t *nv, void *data)
 {
     struct i2c_adapter *pI2cAdapter = (struct i2c_adapter *)data;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
     int osstatus = 0;
     BOOL wasReleased = FALSE;
+#endif
 
 #if defined(KERNEL_2_4)
     if (!NV_WEAK_SYMBOL_PRESENT(i2c_add_adapter))
@@ -324,6 +326,7 @@ BOOL NV_API_CALL nv_i2c_del_adapter(nv_state_t *nv, void *data)
     if (!pI2cAdapter) return FALSE;
 
     // attempt release with the OS
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
     osstatus = i2c_del_adapter(pI2cAdapter);
 
     if (!osstatus)
@@ -333,6 +336,12 @@ BOOL NV_API_CALL nv_i2c_del_adapter(nv_state_t *nv, void *data)
     }
 
     return wasReleased;
+#else
+    i2c_del_adapter(pI2cAdapter);
+    os_free_mem(pI2cAdapter);
+
+    return TRUE;
+#endif
 }
 
 #else // (defined(CONFIG_I2C) || defined(CONFIG_I2C_MODULE))
